frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/NodeAndNPMInstaller.java

```diff
@@ -36,14 +36,14 @@
     }
 
     @Override
-    public void install(String nodeVersion, String npmVersion, String nodeDownloadRoot, String npmDownloadRoot) throws InstallationException {
+    public void install(String nodeVersion, String npmVersion, String nodeDownloadRoot, String npmDownloadRoot, String userName, String password) throws InstallationException {
```

Comment:
These constructor/method signatures keep growing way past their original size; they really need to be refactored to something more appropriate. (It's not this PRs fault, more like the sum of all new functionality added since the class was first written :wink:)

Would you mind changing NodeAndNPMInstaller to be a builder instead? The interface can be deleted, it's not used anyways. We can rename DefaultNodeAndNPMInstaller to just NodeAndNPMInstaller.
How about something like

```java
new NodeAndNPMInstaller()
  .withInstallConfig(...)
  .withArchiveExtractor(...)
  .withFileDownloader(...)
  .withNodeVersion(...)
  .withNpmVersion(...)
  .withNodeDownloadRoot(...)
  .withNpmDownloadRoot(...)
  .withUsername(...)
  .withPassword(...)
  .install();
```